### PR TITLE
PRM-1577 New page for “cover both penalties”

### PR DESCRIPTION
**Pull request by ruthdas03**
Screenshot 2022-07-20 at 14 27 54

### CODE REVIEW

File: app/views/AppealCoverBothPenaltiesPage.scala.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+@this(layout: Layout, pageHeader: components.pageHeader, govukButton: GovukButton)
+
+@()(implicit messages: Messages, appConfig: AppConfig, userRequest: UserRequest[_])
+
+@layout(pageTitle = messages("appealCoverBothPenalties.headingAndTitle"),
+    isPageFullWidth = false,
+    backLinkEnabled = false) {
+        @pageHeader(messages("appealCoverBothPenalties.headingAndTitle"))
+        <p class="govuk-body">@messages("appealCoverBothPenalties.p1")</p>
```
Comment (LewisMitchell) [inline app/views/AppealCoverBothPenaltiesPage.scala.html:27]: Could you unindent this so it lines up with @govuk-button

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
There's no way to get to the page, I know routing is a separate sub-task however we should be able to hit the page manually using a URL. Could you add a controller method and route to show the page?

File: app/controllers/AppealCoverPenaltiesController.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                                                   dataRequired: DataRequiredAction,
+                                                   appConfig: AppConfig,
+                                                   mcc: MessagesControllerComponents) extends FrontendController(mcc) with I18nSupport {
+
+  def onPageLoadForAppealCoverBothPenalties(): Action[AnyContent] = (authorise andThen dataRequired) {
+    implicit request => {
+      Ok(appealCoverBothPenaltiesPage())
+    }
+  }
+}
```
Comment (LewisMitchell) [inline app/controllers/AppealCoverPenaltiesController.scala:39]: Is it easier to use the PenaltySelectionController instead of creating a new one?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: app/controllers/AppealCoverPenaltiesController.scala (hunk 2 above)
Comment (LewisMitchell) [reply app/controllers/AppealCoverPenaltiesController.scala:39]: Maybe that's best because we can don't have lots of single method classes?

Comment (LewisMitchell) [inline app/controllers/AppealCoverPenaltiesController.scala:39]: Do you still need this controller?

File: conf/app.routes [hunk 3]
```diff
@@ -190,3 +190,6 @@ GET        /multiple-penalties-for-this-period/change          controllers.Penal
 
 POST       /multiple-penalties-for-this-period                 controllers.PenaltySelectionController.onSubmitForPenaltySelection(mode: Mode = NormalMode)
 POST       /multiple-penalties-for-this-period/change          controllers.PenaltySelectionController.onSubmitForPenaltySelection(mode: Mode = CheckMode)
+
+# Appeal cover
+GET        /appeal-cover-for-both-penalties                    controllers.AppealCoverPenaltiesController.onPageLoadForAppealCoverBothPenalties()
```
Comment (LewisMitchell) [inline conf/app.routes:195]: This should point to the existing controller (PenaltySelectionController)

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: app/views/AppealCoverBothPenaltiesPage.scala.html [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+@this(layout: Layout,
+      pageHeader: components.pageHeader,
+      link: components.link
+)
+
+@(nextPage: String)(implicit messages: Messages, appConfig: AppConfig, userRequest: UserRequest[_])
+
+@layout(pageTitle = messages("appealCoverBothPenalties.headingAndTitle"),
+    isPageFullWidth = false,
+    backLinkEnabled = false) {
```
Comment (LewisMitchell) [inline app/views/AppealCoverBothPenaltiesPage.scala.html:28]: The back link should be enabled.
I know routing is going to be done in a separate PR, but that should be related to routing to and from these pages, any interaction within these pages should be dealt with now.
An example can be found in: #337

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

**Comment by LewisMitchell** [comment]
Could you also rebase when you get chance?

File: app/views/AppealCoverBothPenaltiesPage.scala.html [hunk 5]
```diff
@@ -21,17 +21,20 @@
       link: components.link
 )
 
-@(nextPage: String)(implicit messages: Messages, appConfig: AppConfig, userRequest: UserRequest[_])
+@(pageMode: PageMode, nextPage: String, penaltyInformation: String)(implicit messages: Messages, appConfig: AppConfig, userRequest: UserRequest[_])
```
Comment (LewisMitchell) [inline app/views/AppealCoverBothPenaltiesPage.scala.html:24]: This page doesn't need penaltyInformation

File: app/models/pages/Page.scala [hunk 6]
```diff
@@ -82,6 +82,8 @@ case object PenaltySelectionPage extends Page
 
 case object AppealSinglePenaltyPage extends Page
 
+case object AppealCoverBothPenaltiesPage extends Page
```
Comment (LewisMitchell) [inline app/models/pages/Page.scala:85]: Could you add this to the allPages val

File: conf/app.routes [hunk 7]
```diff
@@ -187,10 +187,13 @@ GET        /previous-page/change                               controllers.Previ
...
-GET        /appeal-cover-for-both-penalties                    controllers.PenaltySelectionController.onPageLoadForAppealCoverBothPenalties()
+GET        /appeal-cover-for-both-penalties                    controllers.PenaltySelectionController.onPageLoadForAppealCoverBothPenalties(mode: Mode = NormalMode)
+GET        /appeal-cover-for-both-penalties/change             controllers.PenaltySelectionController.onPageLoadForAppealCoverBothPenalties(mode: Mode = CheckMode)
 
 POST       /multiple-penalties-for-this-period                 controllers.PenaltySelectionController.onSubmitForPenaltySelection(mode: Mode = NormalMode)
 POST       /multiple-penalties-for-this-period/change          controllers.PenaltySelectionController.onSubmitForPenaltySelection(mode: Mode = CheckMode)
+POST       /appeal-cover-for-both-penalties                    controllers.PenaltySelectionController.onPageLoadForAppealCoverBothPenalties(mode: Mode = NormalMode)
+POST       /appeal-cover-for-both-penalties/change             controllers.PenaltySelectionController.onPageLoadForAppealCoverBothPenalties(mode: Mode = CheckMode)
```
Comment (LewisMitchell) [inline conf/app.routes:196]: POSTs are only needed if you submit forms

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
I noticed I missed something from my other PR regarding the back link, could you apply the same change here? :sorry:

**Comment by ruthdas03** [comment]
> back link



> I noticed I missed something from my other PR regarding the back link, could you apply the same change here? :sorry:

Is it about enabling the back link, I'm not too sure.

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**